src/app/components/track-list/track-list.component.ts

```diff
@@ -70,22 +71,34 @@ export class TrackListComponent implements OnInit {
     });
   }
 
-  listenTrack(track: ITrack) {
-    this.pauseOtherTracks();
-    track.isPlaying = this.isPlaying;
-    this.playStream(track);
+  playPause(track: ITrack) {
+    if (track.isPlaying) {
+      this.pause(track);
+    } else {
+      this.play(track);
+    }
   }
 
-  pause(trackNumber: number) {
+  pause(track: ITrack) {
+    track.isPlaying = false;
     this.audioService.pause();
-    this.tracks[trackNumber - 1].isPlaying = !this.isPlaying;
   }
 
-  play() {
-    this.audioService.play();
+  play(track: ITrack) {
+    const id = this.audioService.getAudioID();
+    track.isPlaying = true;
+    if (id === track.id) {
+      this.audioService.play();
```

Comment:
I think we should additionally handle case when it is the last audio track. So, add an extra condition **if** it is tracklist end **then** call **this.playStream(track)** to start the track from the beginning along with progress bar.